iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -724,11 +720,7 @@ public override async Task<long> UpdateReportedPropertiesAsync(ReportedPropertie
                     Logging.Info(this, $"Sent twin patch request with request id {requestId}. Now waiting for the service response.");
 
                 // Wait until IoT hub sends a message to this client with the response to this patch twin request.
-                PatchTwinResponse patchTwinResponse = await GetTaskCompletionSourceResultAsync(
-                        taskCompletionSource,
-                        "Timed out waiting for the service to send the updated reported properties version.",
-                        cancellationToken)
-                    .ConfigureAwait(false);
+                PatchTwinResponse patchTwinResponse = await taskCompletionSource.WaitAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
I like this simplification.

Reply:
nit: should we rename this variable to indicate that it is a twinResponseTaskCompletionSource?